Library/Homebrew/extend/cachable.rb

```diff
@@ -11,4 +11,43 @@ def cache
   def clear_cache
     cache.clear
   end
+
+  # Collect all classes that mix in Cachable so that those caches can be cleared in-between tests.
+  if ENV["HOMEBREW_TESTS"]
+    def self.included(klass)
+      raise ArgumentError, "Don't use Cachable with singleton classes" if klass.singleton_class?
```

Comment:
Why does this need to be special-cased/forbidden?

Reply:
I don't think there is a method to get the original class from the singleton class. Forcing classes to extend this module instead of including it in the singleton class avoids that problem and is functionally equivalent.